### DISCUSSION TITLE
Please keep "Trending Repositories and Trending Developers"

### DISCUSSION THREAD

**Discussion by lafin** [community/community#31643] [Other Feature Feedback, Questions, & Ideas] [answered]
Please keep "Trending Repositories and Trending Developers"

![image](https://user-images.githubusercontent.com/1105640/188237560-302f0f29-cc1a-43a1-8a72-640ddbd973fb.png)

**Comment by carlreid** [comment]
I agree! Especially the repositories part.  

I personally find it useful as a little "news feed" in a way I can discover new technologies / ideas. Likewise it can be an indicator of a new major release of a library which I may have otherwise missed.

**Comment by radonn24** [comment]
![trending](https://user-images.githubusercontent.com/13595375/188322051-e4b7e7c4-52aa-4764-a376-82f3d9217070.jpg)

**Reply by LifeIsStrange** [reply]
It's likely decided by a new virtue signaling woke manager who wants their yearly promotion by making a "change".

**Comment by gkapoor121212** [comment]
Please do not remove it!!

**Comment by KrasimirZahariev** [comment]
This is the most useful tab from the Explore family.

**Reply by Conight** [reply]
can't agree more.

**Comment by shajin-sha** [comment]
Do't remove it 😟

**Comment by haringsrob** [comment]
I also like to watch the trending tab almost daily. It helps me discover a lot of new and interesting projects.

**Reply by Kidsunbo** [reply]
same, always a good place to find new repositories

**Reply by juanpabloaj** [reply]
same here, that tab is my daily news

**Comment by EssaAlshammri** [comment]
why would they do that.

plz don't

**Comment by tom-adsfund** [comment]
There's "low usage" and then there's "low usage but also critical to key people". How else will we find the same information? If there's a simple answer then it can go, but otherwise you're removing an essential feature for some people.

**Reply by LifeIsStrange** [reply]
It is very worrying this didn't come to their mind.
Are the people that runs github actually using their own product? Do they understand the implications of their actions?

**Comment by faalatawi** [comment]
This page is very useful for discovering new projects and repos (also old ones 😃 ). It provides a very important way to interact with the platform. Please consider keeping the page.

**Comment by WindowsAddict** [comment]
Please don't remove it. It runs automated, so it's not like you are spending too many resources on maintaining it. 
It helps us to know what people find interesting nowadays. Otherwise how we are supposed to know that?

**Comment by SofianeHamlaoui** [comment]
The Trending Repositories is a good thing to keep, but don't see why they should keep the Developers Trending.

**Reply by LifeIsStrange** [reply]
I don't see why they shouldn't keep the developers trending.. It can be a way to find successful authors and researchers and this trivial code has a nearly zero complexity a and therefore maintenance cost.
It's true it's less critical than trending repos but why do we have to create artificial scarcity in the first place..

**Comment by parkour86** [comment]
I agree with everyone else, we need to keep the trending tab. I use it all the time to find popular projects.

**Comment by matan129** [comment]
I've been using the repositories page every day for a couple of year now, and I'll be really sad to see it go.

It's very useful just to find the current trends - much more concretely than Hacker News, for example.

**Reply by Drawlone** [reply]
me too😭

**Reply by solyasa** [reply]
me too

**Comment by orktes** [comment]
Please keep Trending Repositories. It’s the best way to keep track of the ever changing landscape around different tools and libraries.

**Comment by kkirby** [comment]
What the heck?? I visit trending repos like three times a week. Don't remove it!

**Comment by filipelinhares** [comment]
Please don't remove!

**Comment by Kritiqual** [comment]
I still want the trending repos, not the trending devs. Cause literally every day if I have time, I go to that

**Comment by Gurkengewuerz** [comment]
For me i discover new repositories every day by using the github trending page. Tbh this is where **all** my stars comes from and where I discover new tech stacks. I never used the trending devs to be fair. But please keep the trending repositories!

**Comment by mxhsnowy** [comment]
Please keep both, I check there every morning to see any new technology, trends and recently even repos about life style, old repos sometimes get recognized because of that tab.

**Comment by AlbertVilaCalvo** [comment]
I check it every week, like a clock!

**Comment by loscoala** [comment]
For me it is the most liked feature to discover new ideas and projects.

**Comment by LifeIsStrange** [comment]
I can't believe what my eyes are seeing in front of me.
Removing github trends?!
Has a dumber statement ever been spoken?
I read github trends 5 TIMES PER WEEK SINCE 5 YEARS.
It has allowed me to discover 99.99% of the open source software that I know and is of tremendous value both for my own general culture/curiosity and for my job as a professional developer as there are many trending libraries that push beyond the current limits of software development/computer science.

Removing github trends is synonymous with making your platform undiscoverable and unusable. Do you really think the most popular open source database on earth, AKA your platform, is utterly uninteresting, unimpactful and useless? 

There is a lesson you all have to learn before getting fired, you blind obsessed A/B testers:
*The map is not the territory*
And the observability you believe you have.. You don't.

It's very simple to understand:
Yes it's possible that github trends is used by a relatively low percentage of all users (also consider that I and many block tracking scripts via ublock origin so your stats might be broken)
Even if we assume your stats are not broken, percentages can be misleading as a small percentage of >80 milllions can actually be a lot of people.

**But the most important thing, and that's where there is a remarkably dangerous failure of understanding on your end:**
The supposedly relatively small subset of active users that read github trends has a GIGANTIC snowball effect after the fact, which your statistics cannot observe. 
It is, that, disproportionately, those active users that read github.com trends will be the one to share (on hackernews, reddit, etc) and that will write the actual blogs about the trending libraries. 
It is those either direct links via link aggregators or indirect links via blogs/tweets that will bring the masse visibility of github.com repositories. 
Therefore it is very simple to understand that without giving github.com trends to the active community, there are no longer HN/reddit links nor blogs and the world of open source would become much much less discoverable. And github.com fall into relative irrelevancy/winter. 

You don't want to do that. 

Also, even if statistically few, there are people like me that are polymaths, by becoming meta-expert about libraries in advanced specialized topics I can become the first to see e.g the opportunity of combining a set of innovative libraries to do something that wasn't possible before, such as improving the state of the art in machine learning or even in medecine. 
Removing the access to such free associations of distant knowledges reduce the innovative mental exploration search space and hence induce worldwide scientific regress and is also a direct cause of deaths IRL.


Bonus metaphor:
A program without a main() fails to execute.
A content platform, without an entry point to said content, fails to execute.
Don't remove your entry point.

Please guys, share about the removal, on hackernews, on subreddits, with your github friends, everywhere. We need to fight this.

**Comment by ammikeya** [comment]
Please don't remove! I check it every week, like a clock!

**Comment by xqm32** [comment]
I check it every day and found lots of insteresting repos😭.

Don't remove it please!

**Comment by zmzhang** [comment]
Please keep Trending Repositories and Trending Developers. 
Check them daily and have found many interesting repositories and developers.

**Comment by lizhaode** [comment]
Don't remove it!!!!!!!!!

**Comment by gt2ming** [comment]
Trending is very useful to developers!
Please keep it!

**Comment by 0X0000005** [comment]
keep trending,don't remove,please!

**Comment by shanfl** [comment]
Trending tab is the Window who let me know the Code World

**Comment by jaywcjlove** [comment]
Please do not remove it!!

**Comment by AndrewSpittlemeister** [comment]
This is my primary source for staying on the edge of new tech. I've seriously progressed as a developer by looking through new tech and interesting repositories on this tab. It's also a staple feature of any blog site or social media platform (of which, GitHub is). I think my morning routine will be completely thrown off balance. I'm going to have to picket out front of my office for 5 minute in the morning until it returns. You won't see me, but you'll know I'm there.

**Comment by SnowLee1994** [comment]
I view it everyday, It's a part of my daily life. please do not remove it !!!
I believe it's very useful for many people!

**Comment by itning** [comment]
If it is to be removed, what is the replacement page? Why delete? Is it just low usage? Is this page generation expensive?

**Comment by ifzm** [comment]
I look at it once a day, what are the popular and interesting storage libraries today, I hope not to delete it. This is the most useful feature I think

**Comment by BornZhao** [comment]
PLEASE DON NOT BLIND US

**Comment by JeremyWu917** [comment]
_DO NOT REMOVE IT PLEASE_

**Comment by LitileXueZha** [comment]
Without trendings, no need Github anymore.

**Comment by hunt978** [comment]
while removing trending tab will benefits [libhunt.com](https://www.libhunt.com/). And github.com will be simply repositories for other gold miners  indeed. Money calls~~~

**Comment by hstk30** [comment]
I take a look every day. So please keep it.

**Comment by tredeneo** [comment]
please do not remove it, the repository trends it's good for discover new projects and what's happening ( somes CVE, events and recents projects)

**Comment by 0xlength** [comment]
in my workspace. no trending no github. one of the most important things is to find interesting projects when i using github. I think it is also what developers need.

**Comment by zy6p** [comment]
make trending great again

**Comment by RogerLiNing** [comment]
One of the fun things for me on Github is the `Trending` page, it's the only place where I can get to know what is happening on GitHub. I think it's beneficial, please keep it.

**Comment by LordCasser** [comment]
"One of the fun things for me on Github is the Trending page.",  same for me.

**Comment by princelai** [comment]
plz Do't remove it !

**Comment by DeaglePC** [comment]
That's ridiculous！ You're killing the most popular feature.

**Comment by wuqi** [comment]
That's why  microsoft sucks, they kill everything you like

**Comment by luzizhuo** [comment]
We are working on an open source tool called OSS Insight. https://github.com/pingcap/ossinsight 
Maybe you can consider using it to explore trending repos. https://ossinsight.io/#trending-repos

**Reply by hunt978** [reply]
see, this is what I am talking about.

**Comment by follyxing** [comment]
This is the place that surprises me every day ! Keep it please!

**Comment by pea-cake** [comment]
Please keep

**Comment by elementh** [comment]
Very important feature. Please reconsider.

**Comment by drdh** [comment]
DO NOT REMOVE IT!!!!!!!!

**Comment by Zuckonit** [comment]
so, microsoft no longer leading the trend(ing)

**Reply by hunt978** [reply]
and microsoft killed vcpkg.info site too. they just want you blind and shipping shit to you easily.

**Comment by rui-ktei** [comment]
I actually check trending repositories of programming languages I care about at least once a month. It is one of the main sources I use to learn what's new out there in industry. What's the reasoning behind removing this trending feature?

**Comment by sunxyw** [comment]
+1
Please don't delete it.
I check trending from time to time to find good/exciting projects.

**Comment by billguocs** [comment]
MS should keep it and optimize the trending algo, but not remove it

**Comment by tgiachi** [comment]
Please, keep!
It's part of my morning routine!

**Comment by tekkeitserktok** [comment]
Keep it, I discovered so much useful applications / repositories ! I browse it everyday !

**Comment by dehoej** [comment]
What.. One of my most used features on github.com. My way of getting knowledge of trending technologies.

**Comment by kemics** [comment]
Please, don’t remove, I use it mostly everyday for last 3-4 years

I have found so many great repositories via it

**Comment by melvincwng** [comment]
Both features are pretty useful for me in discovering new repositories or new developers.

Hope this feature can still remain on GitHub. 

🙏🏻

**Comment by pepijnolivier** [comment]
+1 Please don't delete it, not sure what metric is used to mark this as "deprecated due to low usage" - Me and many of my peers use this page frequently

**Comment by pmg1991** [comment]
Please keep "Trending Repositories and Trending Developers"

**Comment by RahulJyala7** [comment]
Please keep this Trending section

**Comment by SimpleIndian** [comment]
Please keep the trending page.
its the best way other developers can explore new OSS Projects

**Comment by Crotalus** [comment]
The person behind this decision should step forward and give an official (preferably honest) explanation why Trending needs to be removed. 

Is it too cumbersome to keep the page running or what?

**Reply by LifeIsStrange** [reply]
Let's not give them the opportunity to bullshit us with false reasons. This page has equivalents in all major popular websites on the internet and github trends has perfectly run for decades.
It's removal is either accidental mediocrity made by an executive that never user their own product, or by malevolence for turning us to dark patterns such as the explore page.
In both case, the person should be fired and public apologies should be said.

**Comment by Crotalus** [comment]
![solved](https://user-images.githubusercontent.com/254972/188201201-1df7d1b6-c257-4300-a030-69f49db51c4c.png)

There you go, the traffic problem solved

**Reply by LifeIsStrange** [reply]
They're a 2 trillion company and you actually did a much better job than them.
It's a tragedy.

**Comment by LifeIsStrange** [comment]
@lafin Hi, can you please integrate this image/message at the end of your first comment?
It shows an obvious and effective solution that would make github.com trends even more useful and impactful than it already is
https://github.com/orgs/community/discussions/31643#discussioncomment-3538615

Doing so would increase visibility

**Comment by mjablecnik** [comment]
I use trending page very often and it is my the most favorite page on GitHub. 
After remove it I probably lost reason why to go here and probably I will migrate to: https://codeberg.org 
which looks as a very nice alternative to GitHub and it is open source, non-profit and community respectful project.

**Comment by lafin** [comment, marked as answer]
https://github.com/orgs/community/discussions/31644#discussioncomment-3539690
